flang/runtime/io-stmt.cpp

```diff
@@ -227,7 +227,18 @@ ConnectionState &ExternalIoStatementBase::GetConnectionState() { return unit_; }
 int ExternalIoStatementBase::EndIoStatement() {
   CompleteOperation();
   auto result{IoStatementBase::EndIoStatement()};
+#if defined(RT_USE_PSEUDO_FILE_UNIT)
+  // Fetch the unit pointer before *this disappears.
+  ExternalFileUnit *unitPtr{&unit_};
+#endif
   unit_.EndIoStatement(); // annihilates *this in unit_.u_
```

Comment:
The comment on this statement isn't relevant on the device.  Maybe it would be more clear to have one `#ifdef / #else / #endif` construct.

Reply:
Thanks, I will change it.